test/components/DateInput_spec.jsx

```diff
@@ -97,6 +97,34 @@ describe('DateInput', () => {
         });
       });
     });
+
+    describe('aria-labelledby', () => {
+      let wrapper;
+      const inputId = 'date';
+      const ariaLabelledBy = 'id1 id2';
+
+      describe('props.ariaLabelledBy is truthy', () => {
+        beforeEach(() => {
+          wrapper = shallow((
+            <DateInput id={inputId} ariaLabelledBy={ariaLabelledBy} />
+          )).dive();
+        });
+
+        it('has aria-labelledby attribute that starts with ariaLabelledBy value', () => {
+          expect(wrapper.find(`input[aria-labelledby="${ariaLabelledBy}"]`)).to.have.lengthOf(1);
```

Comment:
might be good to also validate that the `aria-labelledby` attribute matches the passed in prop value

Reply:
That should be built into this test since the selector wouldn't match on anything that doesn't match the propType, correct?

Reply:
Oh! I missed that you actually had the value specifically passed in (it wrapped in the side by side view and I thought you were just querying for the presence of the attribute). This seems good, and perhaps then it would be good in the second test to check that `aria-labelledby` isn't set at all (and not just that it's not set with the specific value)

Reply:
Yep, that's a better negative test. Will do.

Reply:
So, I added a test that makes sure that `null` is set on `aria-labelledby` in the react props. When rendered, the attribute won't be there in that case. Do you think that's sufficient? Reading around, it seems that testing the action DOM rendering is an antipattern. Thoughts?

Reply:
I don't personally have a strong preference one way or another, but I know some other people do :) I added a suggestion to simplify your existing test to use a similar pattern as to what exists in other tests in this file